robot/rover/internal_comms/src/CommandCenter.cpp

```diff
@@ -1,24 +1,25 @@
 //
 // Created by cedric on 2020-10-14.
+// And half of it was done by Tim
 //
 
 #include "CommandCenter.h"
 #include <etl/queue.h>
 
 namespace internal_comms
 {
-    Command* CommandCenter::processCommand() const
+    Command* CommandCenter::processCommand()
     {
         uint8_t commandID = waitForSerial();
         uint8_t deviceSending = waitForSerial();
         uint8_t deviceReceiving = waitForSerial();
         uint16_t argumentSize = readArgSize();
 
-        uint8_t* buffer = (uint8_t*) malloc(sizeof(uint8_t) * argumentSize);
+        auto* buffer = (uint8_t*) malloc(sizeof(uint8_t) * argumentSize);
```

Comment:
I'd suggest you keep the explicit types and keep `auto` for foreach loops. Make it easier to know what type the variables are if they are explicitly mentioned.

Reply:
They are explicitly mentioned in the cast